Task-57952: Manage "exchange" Agenda Connector

IlhemEssaadi opened the pull request:
After this change, the administrator will be able to manage a new "Exchange" agenda connector by viewing its informations (avatar,  name, description) and disabling/enabling its display in Agenda application and user settings.

boubaker commented on agenda-webapps/src/main/webapp/vue-app/agenda-admin-settings/components/AgendaAdminConnectorSettings.vue:
```diff
@@ -49,7 +49,7 @@
                     <i
                       :key="`icon-${props.item.editing}`"
-                      :class="props.item.editing ? 'uiIcon uiIconTick clickable success--text mt-1' : 'uiIcon uiIconEdit clickable primary--text mt-1'"
+                      :class="props.item.isExchange ? 'd-none' : props.item.editing ? 'uiIcon uiIconTick clickable success--text mt-1' : 'uiIcon uiIconEdit clickable primary--text mt-1'"
```
Why using extensible component plugins if you want to embed the behavior for each connector inside the UI.
IMO, we don't need an extensible framework anymore, we should add isGoogle, isOffice365, isExchange ... The already made work for making this as an extensible framework shouldn't have pieces of plugins. But instead, we should improve the framework to allow different use cases.

IlhemEssaadi replied:
> Why using extensible component plugins if you want to embed the behavior for each connector inside the UI. IMO, we don't need an extensible framework anymore, we should add isGoogle, isOffice365, isExchange ... The already made work for making this as an extensible framework shouldn't have pieces of plugins. But instead, we should improve the framework to allow different use cases.

this is just a draft to synchronize the work with @abenmbarek !

azayati replied:
> You can base on another prop called isOauth for example which allows to decide if the api key is editable or not. This will keep the framework extensible, avoid mentioning exchange in the agenda codebase and prepare the transfert to the agenda-connectors addon codebase

@boubaker is that solution answers to your concern ?

azayati commented on agenda-webapps/src/main/webapp/vue-app/agenda-admin-settings/components/AgendaAdminConnectorSettings.vue:
```diff
@@ -123,6 +125,7 @@ export default {
           connector.apiKey = connectorObj && connectorObj.apiKey || '';
           connector.loading = false;
           connector.editing = false;
+          connector.isExchange = (connector.name === 'agenda.exchangeCalendar') ? true : false;
```
You can base on another prop called isOauth for example which allows to decide if the api key is editable or not. This will keep the framework extensible, avoid mentioning exchange in the agenda codebase and prepare the transfert to the agenda-connectors addon codebase

azayati reviewed (changes requested):
see review

IlhemEssaadi committed:
Task-57952: Add exchange 2016 Calendar Connector (#444)

IlhemEssaadi committed:
Task-57952: Add exchange 2016 Calendar Connector (#445)

Define the text field CSS margin in the case of an exchange connector to match the proposed design.

IlhemEssaadi committed:
Task-57952: Add exchange 2016 Calendar Connector (#448)

Fix new labels for the exchange connector

azayati committed:
Task-57952: Fix sonar check

azayati reviewed (approved):
LGTM
